### PR TITLE
Add docs and tests for `normalize`

### PR DESCRIPTION
**Pull request by Gudahtt**
Inline documentation and tests have been added for the `normalize` function. Only one test was added; the rest were pre-existing. They were grouped together and updated to match our typical testing conventions.

### CODE REVIEW

**Review by rekmarks — CHANGES_REQUESTED** [review]
I imagine the purpose of the `object` test case is to prove to ourselves that inputs that are neither strings nor numbers will be rejected. That being the case, I think we should either note that, or just test a bunch of primitives as well, perhaps by iterating over an array of bad inputs.

File: src/index.ts [hunk 1]
```diff
@@ -353,6 +353,11 @@ function concatSig(v: Buffer, r: Buffer, s: Buffer): string {
   return ethUtil.addHexPrefix(rStr.concat(sStr, vStr)).toString('hex');
 }
 
+/**
+ * Normalize the input to a 0x-prefixed hex string.
+ *
+ * @param input - The value to normalize
```
Comment (Gudahtt) [inline src/index.ts:359]: The return value is missing here

Comment (Gudahtt) [reply src/index.ts]: Resolved in https://github.com/MetaMask/eth-sig-util/pull/178/commits/038f8476225ed1fd0938375ab14a51d9b51f81d0

**Comment by Gudahtt** [comment]
Good idea! I've added tests for all primitive types. I caught some unexpected behaviour for `undefined`, and `null`, so that was interesting.

File: src/index.test.ts [hunk 2]
```diff
@@ -3023,29 +3023,71 @@ describe('concatSig', function () {
...
-    'eth-sig-util.normalize() requires hex string or integer input. received object:',
-  );
+  // TODO: Add validation to disallow negative integers.
+  it('should normalize a negative integer to 0x', function () {
+    const initial = -1;
+    const result = sigUtil.normalize(initial);
+    expect(result).toBe('0x');
+  });
+
+  // TODO: Add validation to disallow undefined.
```
Comment (rekmarks) [inline src/index.test.ts:3052]:
```suggestion
  // TODO: Add validation to disallow null.
```

File: src/index.test.ts [hunk 3]
```diff
@@ -3023,29 +3023,71 @@ describe('concatSig', function () {
...
+  it('should return undefined if given null', function () {
+    const initial = null;
+    expect(sigUtil.normalize(initial as any)).toBeUndefined();
+  });
+
+  // TODO: Add validation to disallow null.
```
Comment (rekmarks) [inline src/index.test.ts:3058]:
```suggestion
  // TODO: Add validation to disallow undefined.
```

File: src/index.test.ts [hunk 4]
```diff
@@ -3023,29 +3023,71 @@ describe('concatSig', function () {
...
+  });
+
+  it('should throw if given a bigint', function () {
+    const initial = BigInt(Number.MAX_SAFE_INTEGER);
+    expect(() => sigUtil.normalize(initial as any)).toThrow(
+      'eth-sig-util.normalize() requires hex string or integer input. received bigint: 9007199254740991',
+    );
+  });
+
+  it('should throw if given a sumbol', function () {
```
Comment (rekmarks) [inline src/index.test.ts:3085]:
```suggestion
  it('should throw if given a symbol', function () {
```

File: src/index.test.ts [hunk 5]
```diff
@@ -3023,29 +3023,71 @@ describe('concatSig', function () {
...
+  });
+
+  it('should throw if given a boolean', function () {
+    const initial = true;
+    expect(() => sigUtil.normalize(initial as any)).toThrow(
+      'eth-sig-util.normalize() requires hex string or integer input. received boolean: true',
+    );
+  });
+
+  it('should throw if given a bigint', function () {
```
Comment (rekmarks) [inline src/index.test.ts:3078]:
```suggestion
  it('should throw if given a BigInt', function () {
```

Comment (Gudahtt) [reply src/index.test.ts]: This one was lower-cased on purpose - apparently the primitive is called `bigint` but the object-wrapper is `BigInt. Kinda like `string` v.s. `String`

**Review by rekmarks — COMMENTED** [review]
LGTM!

File: src/index.test.ts (hunk 2 above)
Comment (Gudahtt) [reply src/index.test.ts]: Whoops! Had those backwards I guess

**Review by rekmarks — APPROVED** [review]
No, it actually LGTM!

File: src/index.test.ts (hunk 5 above)
Comment (Gudahtt) [reply src/index.test.ts]: To reverse this: ```suggestion
  it('should throw if given a bigint', function () {
```

File: src/index.test.ts [hunk 6]
```diff
@@ -3023,29 +3023,71 @@ describe('concatSig', function () {
...
+  });
+
+  it('should throw if given a boolean', function () {
+    const initial = true;
+    expect(() => sigUtil.normalize(initial as any)).toThrow(
+      'eth-sig-util.normalize() requires hex string or integer input. received boolean: true',
+    );
+  });
+
+  it('should throw if given a BigInt', function () {
```
Comment (Gudahtt) [inline src/index.test.ts:3078]: The primitive type is `bigint` apparently
```suggestion
  it('should throw if given a bigint', function () {
```

File: src/index.test.ts (hunk 5 above)
Comment (Gudahtt) [reply src/index.test.ts]: Second attempt: https://github.com/MetaMask/eth-sig-util/pull/178/files#r689795415

**Review by rekmarks — APPROVED** [review]

**Pull request merged**